File: Licenta.Server/DataLayer/Dto/AddIssueDTO.cs
using Licenta.Server.DataLayer.Enum;

namespace Licenta.Server.DataLayer.Dto
{
    public class AddIssueDTO
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime DueDate { get; set; }
        public float EstimatedTime { get; set; }
        public float LoggedTime { get; set; }
        public Guid ProjectId { get; set; }
        public Guid SprintId { get; set; }
        public string AssigneEmail { get; set; }
        public string ReporterEmail { get; set; }
        public string IssueStatus { get; set; }
        public string IssueType { get; set; }
        public Guid ParentIssueId { get; set; }
        public int? LabelId { get; set; }
        public PriorityLevel Priority { get; set; }

    }
}

File: Licenta.Frontend/src/app/Models/DTO/AddIssueDTO.ts
import { PriorityLevel } from "../PriorityLevel";

export class AddIssueDTO
{
    Id: string
    Title: string;
    Description: string;
    CreatedAt: Date;
    DueDate: Date;
    EstimatedTime: number;
    LoggedTime: number;
    ProjectId: string;
    SprintId: string;
    AssigneEmail: string;
    ReporterEmail: string;
    IssueStatus: string;
    IssueType: string; 
    ParentIssueId: string;
    LabelId: number;
    Priority:PriorityLevel;
    constructor()
    {
        this.Id = '';
        this.Title = '';
        this.Description = '';
        this.CreatedAt = new Date();
        this.DueDate = new Date();
        this.EstimatedTime = 0;
        this.LoggedTime = 0;
        this.ProjectId = '';
        this.SprintId = '';
        this.AssigneEmail = '';
        this.ReporterEmail = '';
        this.IssueStatus = '';
        this.IssueType = '';
        this.ParentIssueId = '';
        this.Priority=PriorityLevel.Low;
        this.LabelId=0;
    }
}
